pkg/sip/types.go

```diff
@@ -60,15 +74,38 @@ const (
 type URI struct {
 	User      string
 	Host      string
+	Port      uint16
```

Comment:
If the Host field ends up containing a hostname and port, not an ip address, the Addr argument will be invalid (the preexisting line `netip.AddrPortFrom(u.Addr.Addr(), uint16(port))` will return "Invalid AddPort", with no port field if `u.Addr.Addr()` is invalid).

Reply:
I also thought that it's the case, but it will still work if we test port separately from the address. `addrport.IsValid()` always returns `addrport.Addr().IsValid()`.

But if you think it's inconvenient to use that separately, we could change the field type to `netip.Addr` and add a `AddrPort/SetAddrPort` helper methods.